src/services/renderLoader.ts

```diff
@@ -1,4 +1,10 @@
 import Loading from '../components/Loading';
-import { branch, renderComponent } from 'recompose';
+import { branch, renderComponent, renderNothing, compose } from 'recompose';
 
-export default (dataProp: string = 'data') => branch(props => props[dataProp].loading, renderComponent(Loading));
+export default <T>(dataProp: string = 'data') =>
+    compose(
+        // error handling is handled by the top level App component, so prevent further rendering if errors occur.
+        branch<T>(props => props[dataProp].error, renderNothing),
```

Comment:
Can you make a separate HOC for the branch which checks if it has to render nothing? 
It's not THE fix, but it makes sure that we do not get any bugs in deeper components which do not require a renderLoader after their graphql call